pandas/tests/extension/sparse/test_sparse.py

```diff
@@ -0,0 +1,212 @@
+import pytest
+import pandas as pd
+import numpy as np
+
+from pandas.core.sparse.dtype import SparseDtype
+from pandas import SparseArray
+from pandas.tests.extension import base
+import pandas.util.testing as tm
+
+
+def make_data():
+    data = np.random.uniform(size=100)
+    data[2::3] = np.nan
+    return data
+
+
+@pytest.fixture
+def dtype():
+    return SparseDtype()
+
+
+@pytest.fixture
+def data():
+    """Length-100 PeriodArray for semantics test."""
+    res = SparseArray(make_data())
+    return res
+
+
+@pytest.fixture
+def data_missing():
+    """Length 2 array with [NA, Valid]"""
+    return SparseArray([np.nan, 1.0])
```

Comment:
should we test here one where the fill_value is the missing value, and one where not?